### PR TITLE
chore(tile-converter): i3s-converter - mitigate "tiles" module dependency

### PR DESCRIPTION
**Pull request by belom88**
- removed Tileset3D & Tile3D dependency
- it gives a memory usage improvement (about 10% for Frankfurt dataset)

### CODE REVIEW

File: modules/tile-converter/src/i3s-converter/types.ts [hunk 1]
```diff
@@ -163,3 +163,16 @@ export type TypedArrayConstructor =
   | Uint32ArrayConstructor
   | Float32ArrayConstructor
   | Float64ArrayConstructor;
+
+export type Tiles3DLoadOptions = {
```
Comment (ibgreen) [inline modules/tile-converter/src/i3s-converter/types.ts:167]: Why are these options defined in the tile-converter?

**Review by ibgreen — APPROVED** [review]
I am confused about the goals of this PR and I worry that we may have a misunderstanding.

The goal is not to hide the generic Tile3D from the tile-converter and replace it with a tiles3d specific class. In fact, this approach seems to significantly increase the amount of format specific knowledge that the converter has to deal with, and taking less advantage of the generic `tiles` module.

Regardless there some good typing here - I will approve but wish we had discussed more first.

**Comment by belom88** [comment]
> I am confused about the goals of this PR and I worry that we may have a misunderstanding.
> 
> The goal is not to hide the generic Tile3D from the tile-converter and replace it with a tiles3d specific class. In fact, this approach seems to significantly increase the amount of format specific knowledge that the converter has to deal with, and taking less advantage of the generic `tiles` module.
> 
> Regardless there some good typing here - I will approve but wish we had discussed more first.

We have different classes:
* `I3SConverter` - converts 3DTiles -> I3S
* `Tiles3DConverter` - converts I3S -> 3DTiles.

So it is OK that `I3SConverter` has format specific knowledge.

Initially we decided to get rid of `Tiles3D` and `Tileset3D` to decrease memory usage. The converter creates many instances of `Tiles3D` that is not really needed. For large datasets (100+GiB) it cause significant memory usage growth and even crash. We don't need frustum culling, lod selection and other functionality. Now the converter don't use any wrappers and uses directly output of `Tiles3DLoader`.

File: modules/tile-converter/src/i3s-converter/types.ts (hunk 1 above)
Comment (belom88) [reply modules/tile-converter/src/i3s-converter/types.ts:167]: This is  the i3s-converter specific options

**Comment by belom88** [comment]
TODO: add tests  and TSDoc comments for new code in another PR

File: modules/tile-converter/src/i3s-converter/types.ts (hunk 1 above)
Comment (belom88) [reply modules/tile-converter/src/i3s-converter/types.ts:167]: I've found that we can use `Tiles3DLoaderOptions` here. Fixed in https://github.com/visgl/loaders.gl/pull/2520

**Pull request merged**